addons/sale_management/models/sale_order_template.py

```diff
@@ -29,6 +29,14 @@ def _get_default_require_payment(self):
         domain=[('model', '=', 'sale.order')],
         help="This e-mail template will be sent on confirmation. Leave empty to send nothing.")
     active = fields.Boolean(default=True, help="If unchecked, it will allow you to hide the quotation template without removing it.")
+    company_id = fields.Many2one('res.company', string='Company')
+
+    @api.constrains('company_id', 'sale_order_template_line_ids', 'sale_order_template_option_ids')
+    def _check_company_id(self):
+        for template in self:
+            companies = template.mapped('sale_order_template_line_ids.product_id.company_id') + template.mapped('sale_order_template_option_ids.product_id.company_id')
+            if companies and companies != template.company_id:
```

Comment:
We should check for `template.company_id`
Eg, 
![Screenshot from 2019-07-01 23-31-44](https://user-images.githubusercontent.com/16624719/60456973-74137c80-9c58-11e9-8e64-8dc7c8d9bca1.png)
Cc: @Feyensv

Reply:
Indeed, we should go for sthg like:

```python
if len(companies) > 1 : 
  raise ValidationError(_("Your template cannot contain products from multiple companies"))
  # TODO find better explanation :D
elif companies and template.company_id and companies != template.company_id:
  raise ValidationError(_("Your template contains products from company %s whereas your template belongs to company %s. \n Please change the company of your template or remove the products from other companies.") % (companies.display_name, template.company_id.display_name))
```

Thanks.
(the PR is still in WIP, but feedback's always nice to have ^^).

Reply:
or maybe
```
if len(companies) > 1 : 
  raise ValidationError(_("You cannot have products from multiple companies into one template."))
  
if template.company_id and companies and companies != template.company_id:
  raise ValidationError(_("Your template contains products from company %s whereas your template belongs to company %s. \n Please change the company of your template or remove the products from other companies.") % (companies.display_name, template.company_id.display_name))

```

- First check for `template.company_id`
- elif is not needed here

Does it looks better ?

Reply:
the elif was just to say that we don't raise two errors anyway, but it gives the same result, yes.

Reply:
Well, in the end, we kept it like this and added an onchange to automatically set the company_id of the template if a company-dependent product is added to the template ;).